src/sources/internal_logs.rs

```diff
@@ -34,12 +35,19 @@ impl SourceConfig for InternalLogsConfig {
 }
 
 async fn run(out: Pipeline, mut shutdown: ShutdownSignal) -> Result<(), ()> {
+    let hostname = crate::get_hostname();
     let mut out = out.sink_map_err(|error| error!(message = "Error sending log.", %error));
     let subscription = trace::subscribe();
     let mut rx = subscription.receiver;
 
-    out.send_all(&mut stream::iter(subscription.buffer).map(Ok))
-        .await?;
+    out.send_all(&mut stream::iter(subscription.buffer).map(|mut log| {
+        if let Ok(hostname) = &hostname {
+            log.insert(log_schema().host_key().to_owned(), hostname.to_owned());
+        }
+        log.insert(String::from("pid"), std::process::id());
```

Comment:
If we don't expect our PID to change, this could get hoisted up like hostname.

Reply:
Doh, yeah, thanks. I meant to do this after rebasing.